src/gpuarray_buffer_cuda.c

```diff
@@ -290,7 +270,48 @@ static void *cuda_init(int ord, int flags, int *ret) {
 
     return res;
 }
+static void *cuda_init(int ord, int flags, int *ret) {
+    CUdevice dev;
+    cuda_context *res;
+    static int init_done = 0;
 
+    if (ord == -2) {
+      CUcontext ctx;
+      /* Grab the ambient context */
+      err = cuCtxGetCurrent(&ctx);
+      CHKFAIL(NULL);
+      res = cuda_make_ctx(ctx, DONTFREE);
+      if (res == NULL) {
+        FAIL(NULL, GA_IMPL_ERROR);
+      }
+      res->flags |= flags;
+      return res;
```

Comment:
I though that would be implied by the `/* Grab the ambient context */` comment.  It's kind of impossible to have a context without initializing the API.

What I was referring to was that you were probably right and that we probably should set init_done to 1 even if we didn't call cuInit() ourselves since it should have been done before anyway.